draft-ietf-quic-http.md

```diff
@@ -585,6 +585,116 @@ The payload consists of:
+as though they had never been sent at all, thereby allowing those requests to be
+retried later on a new connection.  Servers SHOULD treat requests on streams
+above this ID as stream errors of type HTTP_REQUEST_CANCELLED.
+
+Endpoints SHOULD send a GOAWAY frame when the closing of a connection is known
+in advance, even if the window is small, so that the remote peer can know
+whether a stream has been partially processed or not.  For example, if an HTTP
+client sends a POST at the same time that a server closes a QUIC connection, the
+client cannot know if the server started to process that POST request if the
+server does not send a GOAWAY frame to indicate what streams it might have acted
+on.
+
+For unexpected closures caused by error conditions, transport-level mechanisms
+SHOULD be used.  However, a GOAWAY MAY still be sent to provide additional
+detail to clients.
+
+~~~~~~~~~~  drawing
+ +---------------------------------------------------------------+
+ |                   Last-Stream-ID (32)                         |
+ +---------------------------------------------------------------+
+ |                     [Error Code (32)]                         |
+ +---------------------------------------------------------------+
+ |                 [Additional Debug Data (*)]                 ...
+ +---------------------------------------------------------------+
+~~~~~~~~~~
+ {: #fig-goaway title="GOAWAY frame payload"}
+
+The GOAWAY frame does not define any flags.
+
+The GOAWAY frame applies to the connection, not a specific stream. An endpoint
+MUST treat a GOAWAY frame on a stream other than the control stream as a
+connection error ({{errors}}) of type HTTP_WRONG_STREAM.
+
+The last stream identifier in the GOAWAY frame contains the highest-numbered
+stream identifier for which the sender of the GOAWAY frame might have taken some
+action on or might yet take action on.  All streams up to and including the
+identified stream might have been processed in some way.  The last stream
+identifier can be set to 0 if no streams were processed.
+
+> Note: In this context, "processed" means that some data from the stream was
+> passed to some higher layer of software that might have taken some action as a
+> result.
+
+If a connection terminates without a GOAWAY frame, the last stream identifier is
+effectively the highest possible stream identifier.
+
+Reattempting requests on streams with lower- or equal-numbered identifiers is
+not possible, with the exception of idempotent actions like HTTP GET, PUT, or
+DELETE.  Any protocol activity that used higher-numbered streams can be safely
+retried using a new connection.
+
+Activity on streams numbered lower or equal to the last stream identifier might
+still complete successfully.  The sender of a GOAWAY frame might gracefully shut
+down a connection by sending a GOAWAY frame, maintaining the connection in an
+"open" state until all in-progress streams complete.
```

Comment:
We can be more concrete about this.  Say that GOAWAY doesn't cause the connection to close, and the connection will remain open until requests on streams up to the last stream ID have completed.  After that, the connection can be closed (let's avoid making a statement about silent close vs. a signaled close at this point by remaining silent on exactly *how* it is closed).